# PART 1

**GENE-BASE ANALYSIS WITH MAGMA**

#This tutorial assumes that MAGMA and all the files required for the tutorial are in the same directory. If your files are in different directories you must add the directory path to the command line. To begin set your working directory and unzip the files

    cd /path/to-your folder.

#Unzip the program folders and the data file magma_v1.07b.zip, NCBI37.3.zip and MDD2018_excluding23andMe_short (downland this file from the PGC website)
  
    unzip *file.zip 

**Annotation* Map SNPs to genes**

#Annotation is preliminary to the gene association analysis, in this step SNPs from the GWAS summary statistics are mapped to genes in the gene location file. For convenience we start by extracting SNP ID, chromosome and base pair position from the GWAS summary, into a snp-loc file in MAGMA format, we also extracted the p-value information which will be used in a later step.
  
    awk '{print $2,$1,$3,$11}' MDD2018_excluding23andMe > MDD2018_excluding23andMe_short.txt

#It is necessary that the input summary data is organized with the three first columns matching the order: SNP ID, chromosome and base pair position, MAGMA ignores other columns. To run the annotation use the code below, it will generate a genes.annot [MDD_list.genes.annot] file with all SNPs that mapped to a gene and .log file [MDD_list.log] that should be inspected for warning and errors. 

    ./magma --annotate --snp-loc MDD2018_excluding23andMe_short.txt 
      --gene-loc NCBI37.3.gene.loc 
      --out MDD_list
    
**GENE-BASED ANALYSIS ON SNP-PVALUE DATA**

#In this step SNPs are assigned to nearest gene and gene-based statistics are computed based on the sum of SNP-log (10) P-values. The analysis requires of raw genotype data, P-value information, sample size and the gene annotation file generated in the previous step. For this tutorial the raw genotype data correspond to the 1000 genome reference file for European population [g1000_eur], P-values were extracted from the MDD GWAS summary data prepared in the previous step [MDD2018_excluding23andMe_short.txt]. The total number of samples for the MDD GWAS is N=307,354. Multiple testing is done using 10000 adaptative permutations (--adap-permp=10000). The output is a genes.out file and a genes.raw file that would be used in the next step.

    ./magma --bfile g1000_eur  
      --gene-annot MDD_list.genes.annot 
      --pval MDD2018_excluding23andMe_short.txt N=307,354  
      --gene-settings adap-permp=10000 
      --out magma
    
   **GENE_SET ANALYSIS**

#This step runs a competitive gene-set analysis as described in ref(xx). Here we test for the association between groups of genes within a tissue and the phenotypes i.e. MDD. The analysis counts for gene size and gene density. The gene sets used in this tutorial are the networks files [normalised.tx] that were generated by refxxxx (xxx or maybe something like details of how the networks were build are provided in xxx) and are freely provided. Each file correspond to genes within a human tissue and gene sets are defined by colours (refxxx). Each file have 2 columns: set and gene ID. The second input on the gene-set anlaysis is the .raw outcome from the gene-based analysis [MDD_genes.raw]. If a gene-set within a network has a significant association with the phenotype the code below outputs a gsa.sets.genes.out file, a magma.gsa.genes.out file and a gsa.outoutputs. For the gene networks that don't have significant association only the gsa.out file is output.

    for file in network_files/*_normalised.txt; do 
    ./magma --gene-results MDDasso_magma.genes.raw 
    --set-annot "$file" col=2,1 
    --out "$file"_MDD_magma
  
The code above outputs one significant gene-set file for Thyroid:
        
        head Thyroid_entrez_gtex_v7_normalised.txt_MDD_permuta_magma.gsa.sets.genes.out 
        
 The file looks like this:
        
        # ALPHA = 0.05
        # NUMBER_OF_TESTS = 27

        # _SET1_  VARIABLE = saddlebrown (set)
        # _SET1_  NGENES = 1072
        # _SET1_  P-VALUE = 0.000796434
        _SET1_   GENE        CHR      START       STOP  NSNPS  NPARAM    N        ZSTAT            P  ZFITTED_BASE  ZRESID_BASE
        _SET1_   26155         1     879583     894679     95      21  307      0.73902      0.21525    0.00033225      0.73869
        _SET1_   51150         1    1152288    1167447    126      10  307     -0.38593      0.59927    0.00033225     -0.38626
        _SET1_   54998         1    1309110    1310818      9       5  307        1.237      0.10463    0.00033225       1.2367

        
The files has information on the significant sets: out of 27 gene-sets MDD associations signals were significanlty enriched in gene-set saddlebrown, with a P-value=0.000796434, this set has 1072 genes, listed in the file. Using grep we can extract the beta and SE values for the significant sets from the gsa.out file. 

    grep saddlebrown Thyroid_entrez_gtex_v7_normalised.txt_MDD_permuta_magma.gsa.out

To filter out the top associated the genes with a treshold of 0.05 we can do:

    sort -n -r -k10 Thyroid_entrez_gtex_v7_normalised.txt_MDD_permuta_magma.gsa.sets.genes.out|awk '{if ($10<=0.05) print $0}' > top_thyroid_MDD_saddlebrown

Thi will generated a list of 133 genes.

So far we generated a list of genes that based on gene-proximity are associated with MDD. In the second part of this tutorial we will use eMAGMA to generate a list of genes associated to MDD based on genes function go to PART2!
